### ISSUE TITLE
videograbber: reinitialize USB device if there were no videos connected for e.g. 24 h

### ISSUE THREAD

**Issue by yarikoptic** [ReproNim/reprostim#123]
In relation to #121 

notes:
- should not reset if there is ongoing video collection

**Comment by yarikoptic** [comment]
FWIW, I just stopped by and 
- it didn't work through HDMI -> HDMI2VGA -> VGA F2F  adapter
- worked just fine when I connected via USB... 
 
so mystery remains somewhat on why   it did not work 2 days ago for me

**Comment by vmdocua** [comment]
`Status of the Blue Indicator Light` in Magewell USB Capture+ device:

- `On`: The indicator light will be on steadily when a signal source is connected.
- `Breathing`: The indicator light is pulsing slowly when there is no signal, the signal cannot be locked or supported.
- `Flashing`: The indicator light flashes when there is a problem with the device.

Most likely device was in sleep/breathing mode based on our discussion. There is no direct API in SDK to keep device always up, but videocapture code permanently in cycle 1 time /sec invokes device API  like `MWOpenChannelByPath`/`MWGetVideoSignalStatus`, theoretically it should keep it up. In addition to device sleep mode we should investigate provided power and USB power management settings on host computer, to make sure that box doesn't switch device into a low-power state when idle.  

To trace/monitor `reprostim-videocapture` issues we can use it with `--versbose` command-line option, but it produces large logs. Looked at it, but didn't find something unisuall.

When stopping `reprostim-videocapture` it also good to make sure that no capture or `ffmpeg` processes accidentaly up, like:
```shell
ps aux | grep videocapture
ps aux | grep ffmpeg
```

May be we also need to install `USB Capture Utility` at this box, so we can start it manually for debug/monitor purposes (https://www.magewell.com/downloads/usb-capture#/tools/linux-ubuntu). 

Other option there is to run UI and see how VLC can capture video signal visually.

P.S.: After box was restarted somehow `repromon` conatiners were down. It has been restarted, so now it should work fine.

**Comment by vmdocua** [comment]
Some notes based on device behaviour and `reprostim-videocapture` process automatic restart (via config reload) or manuall direct process restart and fact it didn't help in this situation. 

Proposal in this area: As we have `repromon` and capture utility connected to it. We can detect inactivity for some long period of time (like 24h+-) and send `ERROR` status message to `repromon` system in this case. Just my 2c.

**Comment by yarikoptic** [comment]
> Proposal in this area: As we have `repromon` and capture utility connected to it. We can detect inactivity for some long period of time (like 24h+-) and send `ERROR` status message to `repromon` system in this case. Just my 2c.

Good idea, but ideally/eventually it should be even more precise as it should be ERRORed if there were MRI acquisitions in the day but no videos recorded.  Ideally if there is MRI with `func` prefix and not `task-rest` -- then definitely there should have been a video! ;)  May be I should first adjust my cron job (the `tools/cron-check-reproiner-videos`) to perform that to see how well that rule applies!  But with that in mind -- unlikely it should be the job of `reprostim-videocapture` to send ERROR.

**Comment by yarikoptic** [comment]
re the last comment, pushed that tune up to cron script in http://github.com/ReproNim/reprostim/commit/baf076dcfd57f9b157b2a358f928f8f6e67baadd

**Comment by vmdocua** [comment]
> Good idea, but ideally/eventually it should be even more precise as it should be ERRORed if there were MRI acquisitions in the day but no videos recorded. Ideally if there is MRI with `func` prefix and not `task-rest` -- then definitely there should have been a video! ;) May be I should first adjust my cron job (the `tools/cron-check-reproiner-videos`) to perform that to see how well that rule applies! But with that in mind -- unlikely it should be the job of `reprostim-videocapture` to send ERROR.

This is true that should exist some external monitor or watchdog to control video capture status. ATM `reprostim-videocapture` sends to `repromon` system some high level messages under `ReproStim` category for informational purposes (like utility started/terminated or capture session started/finished or USB device connected/disconnected etc) .
